.changeset/ten-months-kick.md

```diff
@@ -0,0 +1,5 @@
+---
+"@clerk/nextjs": minor
+---
+
+Allows access to request object to dynamically define `clerkMiddleware` options
```

Comment:
I'll wait on merging this [docs PR](https://github.com/clerk/clerk-docs/pull/1508/files) to reference it in the changeset here